core.md

```diff
@@ -177,84 +249,105 @@ initialized to 0, fields of type `any` are initialized to `undefined`,
 and so on). Fields with structural type (like `from` and `to` in this
 case) are recursively initialized.
 
-When creating a new typed object, you can also supply an "example
+When creating a new typed object, you can also supply a "source
 object". This object will be used to extract the initial values for
 each field:
 
 ```js
-var line1 = new LineType({from: {x: 1, y: 2},
+let line1 = new LineType({from: {x: 1, y: 2},
                           to: {x: 3, y: 4}});
 console.log(line1.from.x); // logs 1
 
-var line2 = new LineType(line1);
+let line2 = new LineType(line1);
 console.log(line2.from.x); // also logs 1
-
-var PointsType = PointType.array(2);
-var array = new PointsType([{x: 1, y: 2}, {x: 3, y: 4}]);
-console.log(array[0].x); // ALSO logs 1
 ```
 
 As the example shows, the example object can be either a normal JS
 object or another typed object. The only requirement is that it have
-fields (or elements, in the case of an array) of the appropriate
-type. Essentially, writing:
+fields of the appropriate type. Essentially, writing:
 
 ```js
-var line1 = new LineType(example);
+let line1 = new LineType(example);
 ```
 
 is exactly equivalent to writing:
```

Comment:
In this particular example it is, but I clarified that there are caveats to this equivalence (with or without the ability to specify defaults).